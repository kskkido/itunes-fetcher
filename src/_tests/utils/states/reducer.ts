import { Action } from 'redux';
import { rootReducer, RootState } from 'states/root/reducer';
import { noop } from './action';

// TODO fix types
export const batchedActionsReducer = (
  state: RootState | {},
  actions: Action[] = [noop]
) => (
  actions.reduce(
    rootReducer,
    state as RootState
  ) as RootState
);
