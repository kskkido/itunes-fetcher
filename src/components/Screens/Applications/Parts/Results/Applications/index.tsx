import React from 'react';
import Section from 'components/Shared/Section';
import ListColumn from 'components/Shared/ListColumn';
import Application from 'containers/Screens/Applications/Parts/Results/Application';

type ApplicationsProps = {
  ids: string[];
};

const Applications: React.SFC<ApplicationsProps> = ({
  ids,
}) => (
  <Section>
    <ListColumn>
      {ids.map(id => <Application key={id} id={id} />)}
    </ListColumn>
  </Section>
);

export default Applications;
