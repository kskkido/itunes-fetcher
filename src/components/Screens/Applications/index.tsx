import React from 'react';
import PageContainer from 'components/Shared/Layouts/PageContainer';
import ResultsApplications from 'containers/Screens/Applications/Parts/Results/Applications';

const Applications: React.SFC = () => (
  <PageContainer>
    <ResultsApplications />
  </PageContainer>
);

export default Applications;
