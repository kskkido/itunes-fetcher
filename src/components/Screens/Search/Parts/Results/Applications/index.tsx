import React from 'react';
import { NAVIGATION_SEARCH_APPLICATIONS } from 'routes/navigations';
import Link from 'components/Shared/Link';
import Heading from 'components/Shared/Heading';
import Section from 'components/Shared/Section';
import SectionHeader from 'components/Shared/SectionHeader';
import ListColumn from 'components/Shared/ListColumn';
import Application from 'containers/Screens/Search/Parts/Results/Application';

type ApplicationsProps = {
  ids: string[];
};

const Applications: React.SFC<ApplicationsProps> = ({
  ids,
}) => (
  <Section>
    <SectionHeader>
      <Heading level={3}>
        Apps
      </Heading>
      <Link to={NAVIGATION_SEARCH_APPLICATIONS.name}>
        See All...
      </Link>
    </SectionHeader>
    <ListColumn>
      {ids.map(id => <Application key={id} id={id} />)}
    </ListColumn>
  </Section>
);

export default Applications;
