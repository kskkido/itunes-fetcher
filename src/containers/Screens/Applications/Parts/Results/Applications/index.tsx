import React from 'react';
import { useSelector } from 'react-redux';
import * as applicationsSelector from 'states/applications/selectors';
import Render from 'components/Shared/Render';
import ApplicationsComponent from 'components/Screens/Applications/Parts/Results/Applications';

const Applications: React.SFC = () => {
  const ids = useSelector(applicationsSelector.makeSelectAllIds());

  return (
    <Render when={ids.length > 0}>
      <ApplicationsComponent ids={ids} />
    </Render>
  );
};

export default Applications;
