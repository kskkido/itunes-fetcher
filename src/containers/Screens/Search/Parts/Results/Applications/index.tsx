import React from 'react';
import { useSelector } from 'react-redux';
import * as applicationsSelectors from 'states/applications/selectors';
import Render from 'components/Shared/Render';
import ApplicationsComponent from 'components/Screens/Search/Parts/Results/Applications';

const Applications: React.SFC = () => {
  const ids = useSelector(applicationsSelectors.makeSelectAllIds());

  return (
    <Render when={ids.length > 0}>
      <ApplicationsComponent ids={ids.slice(0, 4)} />
    </Render>
  );
};

export default Applications;
