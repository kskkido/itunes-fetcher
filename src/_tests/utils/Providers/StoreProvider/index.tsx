import React from 'react';
import { Provider } from 'react-redux';
import { RootState } from 'states/root/reducer';
import { configureStore } from 'states';

type StoreProviderProps = {
  children: React.ReactNode;
  state: Partial<RootState>;
};

const StoreProvider: React.SFC<StoreProviderProps> = ({
  children,
  state,
}) => (
  <Provider store={configureStore(state)}>
    {children}
  </Provider>
);

export default StoreProvider;
