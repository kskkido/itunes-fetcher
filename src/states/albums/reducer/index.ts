import { combineReducers, asClearableReducer } from 'states/utils/reducer';
import { clear } from 'states/albums/actions';
import { recordsReducer, RecordsState } from './records';
import { requestReducer, RequestState } from './request';

export type AlbumsState = {
  records: RecordsState;
  request: RequestState;
};

export const albumsReducer = asClearableReducer(
  combineReducers<AlbumsState>({
    records: recordsReducer,
    request: requestReducer,
  }),
  clear.type
);
