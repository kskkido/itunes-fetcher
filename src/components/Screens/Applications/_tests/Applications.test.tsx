/// <reference types="jest" />
import React from 'react';
import TestRenderer from 'react-test-renderer';
import StoreProvider from '_tests/utils/Providers/StoreProvider';
import { batchedActionsReducer } from '_tests/utils/states/reducer';
import Applications from '../';

const setup = () => ({
  tree: TestRenderer.create(
    <StoreProvider state={batchedActionsReducer({})}>
      <Applications />
    </StoreProvider>
  ),
});

describe('Applications', () => {
  it('should render as expected', () => {
    expect(setup().tree.toJSON()).toMatchSnapshot();
  });
});
