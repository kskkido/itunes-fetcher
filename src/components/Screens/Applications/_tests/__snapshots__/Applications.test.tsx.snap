// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications should render as expected 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View />
</RCTScrollView>
`;
